### PR TITLE
Update host_filter.rb

### PR DESCRIPTION
**Pull request by cseeger-epages**
possible fix for #1766

### CODE REVIEW

**Comment by will-in-wi** [comment]
The test failure is Rubocop carping about using `/regex/` instead of `%r{regex}` in this particular case. You should be able to run `bundle exec rubocop -a` and it'll fix the issue for you.

Could you add a couple tests for this change? There are existing tests in `spec/lib/capistrano/configuration/role_filter_spec.rb`. I'd just add one which tries to match `server`. It would return all of them before this fix and none after.

Thanks!

**Comment by will-in-wi** [comment]
Could you also add a changelog entry and give yourself credit?

**Comment by cseeger-epages** [comment]
i will take a look at the tests now

**Comment by will-in-wi** [comment]
Thanks for working on this!

**Comment by will-in-wi** [comment]
👍 I think this looks good. @mattbrictson: merge?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ https://github.com/capistrano/capistrano/compare/v3.6.1...HEAD
 
 * Your contribution here!
 * `remote_file` feature has been removed and is no longer available to use @SaiVardhan
+* fix bug with greedy matching in host_filter (@cseeger-epages)
```
Comment (mattbrictson) [inline CHANGELOG.md]: Could you also reference the issue number? E.g.

```
[#1766](https://github.com/capistrano/capistrano/issues/1766)
```

**Comment by mattbrictson** [comment]
Please see me line comment regarding the changelog. Otherwise this looks good. Thanks!

**Comment by mattbrictson** [comment]
Thank you!

**Pull request merged**